src/bin/kawari-world.rs

```diff
@@ -402,6 +402,53 @@ async fn client_loop(
                                                         }
 
                                                         connection.handle.send(ToServer::ZoneIn(connection.id, connection.player_data.actor_id, connection.player_data.teleport_reason == TeleportReason::Aetheryte)).await;
+                                                    },
+                                                    ClientTriggerCommand::BeginContentsReplay {} => {
+                                                        connection.conditions.set_condition(Condition::ContentsReplay);
+                                                        connection.send_conditions().await;
+
+                                                        connection.actor_control_self(ActorControlSelf {
+                                                            category: ActorControlCategory::BeginContentsReplay {
+                                                                unk1: 1
+                                                            }
+                                                        }).await;
+                                                    },
+                                                    ClientTriggerCommand::EndContentsReplay {} => {
+                                                        connection.actor_control_self(ActorControlSelf {
+                                                            category: ActorControlCategory::EndContentsReplay {
+                                                                unk1: 1
+                                                            }
+                                                        }).await;
+
+                                                        // TODO: de-duplicate from ClientZoneIpcData::FinishLoading
+                                                        let common = connection.get_player_common_spawn(connection.exit_position, connection.exit_rotation);
```

Comment:
This isn't a nitpick of your code (no, it looks great - thanks!) but more of a i-never-used-duty-recorder-so-im-stupid comment. I guess it takes over the game state, so this is just to spawn your actual player back in once the replay is done right?

Reply:
Thank you! And yes, the ActorControlSelf basically brings you back to "reality", and although the zone it's still loaded on the client, it needs a PlayerSpawn to bring back the player. And then, the Condition clear is needed because after spawning, the player is locked and can't do anything.